File: app/src/main/java/arun/com/chromer/activities/intro/AppIntroFragment.java
package arun.com.chromer.activities.intro;

/**
 * Created by Arun on 06/02/2016.
 */

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.github.paolorotolo.appintro.R.id;
import com.github.paolorotolo.appintro.R.layout;

public class AppIntroFragment extends Fragment {
    private int drawable;
    private int bgColor;
    private int titleColor;
    private int descColor;
    private CharSequence title;
    private CharSequence description;

    public AppIntroFragment() {
    }

    public static AppIntroFragment newInstance(CharSequence title, CharSequence description, int imageDrawable, int bgColor) {
        return newInstance(title, description, imageDrawable, bgColor, 0, 0);
    }

    @SuppressWarnings("WeakerAccess")
    public static AppIntroFragment newInstance(CharSequence title, CharSequence description, int imageDrawable, int bgColor, int titleColor, int descColor) {
        AppIntroFragment sampleSlide = new AppIntroFragment();
        Bundle args = new Bundle();
        args.putCharSequence("title", title);
        args.putCharSequence("desc", description);
        args.putInt("drawable", imageDrawable);
        args.putInt("bg_color", bgColor);
        args.putInt("title_color", titleColor);
        args.putInt("desc_color", descColor);
        sampleSlide.setArguments(args);
        return sampleSlide;
    }

    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (this.getArguments() != null && this.getArguments().size() != 0) {
            this.drawable = this.getArguments().getInt("drawable");
            this.title = this.getArguments().getCharSequence("title");
            this.description = this.getArguments().getCharSequence("desc");
            this.bgColor = this.getArguments().getInt("bg_color");
            this.titleColor = this.getArguments().containsKey("title_color") ? this.getArguments().getInt("title_color") : 0;
            this.descColor = this.getArguments().containsKey("desc_color") ? this.getArguments().getInt("desc_color") : 0;
        }

    }

    @Nullable
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(layout.fragment_intro, container, false);
        TextView t = (TextView) v.findViewById(id.title);
        TextView d = (TextView) v.findViewById(id.description);
        ImageView i = (ImageView) v.findViewById(id.image);
        LinearLayout m = (LinearLayout) v.findViewById(id.main);
        t.setText(this.title);
        if (this.titleColor != 0) {
            t.setTextColor(this.titleColor);
        }

        d.setText(this.description);
        if (this.descColor != 0) {
            d.setTextColor(this.descColor);
        }

        // Use glide to load the drawable
        Glide.with(this).load(drawable).crossFade().into(i);

        m.setBackgroundColor(this.bgColor);
        return v;
    }
}

File: android-app/lynket/src/main/java/arun/com/chromer/intro/fragments/AppIntroFragment.kt
/*
 *
 *  Lynket
 *
 *  Copyright (C) 2022 Arunkumar
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package arun.com.chromer.intro.fragments

import android.os.Bundle
import androidx.annotation.ColorInt
import androidx.annotation.DrawableRes
import androidx.core.content.ContextCompat
import arun.com.chromer.R
import arun.com.chromer.di.fragment.FragmentComponent
import arun.com.chromer.shared.base.fragment.BaseFragment
import arun.com.chromer.util.glide.GlideApp
import com.github.paolorotolo.appintro.ISlideBackgroundColorHolder
import kotlinx.android.synthetic.main.fragment_text_intro.*

open class AppIntroFragment : BaseFragment(), ISlideBackgroundColorHolder {

  override fun getDefaultBackgroundColor(): Int =
    ContextCompat.getColor(context!!, R.color.tutorialBackgrounColor)

  override fun setBackgroundColor(backgroundColor: Int) {
    if (root != null) {
      root.setBackgroundColor(backgroundColor)
    }
  }

  override fun inject(fragmentComponent: FragmentComponent) = fragmentComponent.inject(this)

  override val layoutRes: Int get() = R.layout.fragment_text_intro

  private var drawable: Int = 0
  private var bgColor: Int = 0
  private var titleColor: Int = 0
  private var descColor: Int = 0
  private var title: CharSequence? = null
  private var description: CharSequence? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    if (this.arguments != null && this.arguments!!.size() != 0) {
      this.drawable = this.arguments!!.getInt("drawable")
      this.title = this.arguments!!.getCharSequence("title")
      this.description = this.arguments!!.getCharSequence("desc")
      this.bgColor = this.arguments!!.getInt("bg_color")
      this.titleColor =
        if (this.arguments!!.containsKey("title_color")) this.arguments!!.getInt("title_color") else 0
      this.descColor =
        if (this.arguments!!.containsKey("desc_color")) this.arguments!!.getInt("desc_color") else 0
    }
  }

  override fun onActivityCreated(savedInstanceState: Bundle?) {
    super.onActivityCreated(savedInstanceState)
    root.setBackgroundColor(this.bgColor)
    titleTv.text = this.title
    if (this.titleColor != 0) {
      titleTv!!.setTextColor(this.titleColor)
    }

    descriptionTv.text = this.description
    if (this.descColor != 0) {
      descriptionTv!!.setTextColor(this.descColor)
    }

    // Use glide to load the drawable
    GlideApp.with(this).load(drawable).into(imageView!!)
  }

  companion object {
    fun newInstance(
      title: CharSequence,
      description: CharSequence,
      @DrawableRes imageDrawable: Int,
      @ColorInt bgColor: Int
    ): AppIntroFragment {
      return newInstance(title, description, imageDrawable, bgColor, 0, 0)
    }

    fun newInstance(
      title: CharSequence,
      description: CharSequence,
      @DrawableRes imageDrawable: Int,
      @ColorInt bgColor: Int,
      @ColorInt titleColor: Int,
      @ColorInt descColor: Int
    ): AppIntroFragment {
      val sampleSlide = AppIntroFragment()
      val args = Bundle()
      args.putCharSequence("title", title)
      args.putCharSequence("desc", description)
      args.putInt("drawable", imageDrawable)
      args.putInt("bg_color", bgColor)
      args.putInt("title_color", titleColor)
      args.putInt("desc_color", descColor)
      sampleSlide.arguments = args
      return sampleSlide
    }
  }
}
